### ISSUE TITLE
Support for WinUI3.0 and Net 5.0

### ISSUE THREAD

**Issue by andres-gimenez** [MvvmCross/MvvmCross#3816]
## 🚀 Feature Requests

Will WinUI3.0 be deployed with Net5.0 on MvvmCross 7?

### Platforms affected (mark all that apply)
- [ ] :iphone: iOS
- [ ] :robot: Android
- [x] :checkered_flag: WPF
- [x] :earth_americas: UWP
- [ ] :apple: MacOS
- [ ] :tv: tvOS
- [ ] :monkey: Xamarin.Forms

**Comment by Cheesebaron** [comment]
Not in the first version of MvvmCross 7. Unless you have something ready right now. We are open for contributions though!

**Comment by andrechi1** [comment]
> Not in the first version of MvvmCross 7. Unless you have something ready right now. We are open for contributions though!

I tested a build with net5.0. But I've come across trouble. 

- Now MSBuild.Sdk.Extras doesn't support Net5.0. 
- WPF and UWP are unified. But I don't know how.

Now it's not well known as Net5.0 will be but I want to know how the migration will be done. By platforms or all together.

If I can help. I'm delighted.

**Comment by pinkysek** [comment]
+1 for support WinUI 3.0, it looks cool.

**Comment by bkaankose** [comment]
I tried to convert my existing UWP app (using WinUI 2.5 + MvvmCross) to WinUI UWP 3. Things didn't go well.

**My observations**
- Anything  inherits Windows.UI.Xaml.Controls should be changed to Microsoft.UI.Xaml.Controls. Including **MvxWindowsPage**
- MvxCommand and MvxAsyncCommand inherits System.Windows.Input.ICommand. However, WinUI3 controls expect [XamlUICommand](https://docs.microsoft.com/en-us/windows/winui/api/microsoft.ui.xaml.input.xamluicommand?view=winui-3.0-preview) There must be either some converter to use it in XAML, or a new class implementation.
- Same goes for[ INotifyCollectionChanged and INotifyPropertyChanged (Check known issues)](https://docs.microsoft.com/en-us/windows/apps/winui/winui3/) . They have different namespaces in WinUI3 Preview.
- UWP and Win32 worlds are quite different now in WinUI3. Unification still continues, but based on the current implementation and targeting options it is not possible to do it in MvvmCross with less effort.
- Anything dialog related is not fully supported in WinUI3. For example ContentDialog because it binds to PopupRoot where it's not available in WinUI Visual Tree yet. MvvmCross's view presenter for UWP should be aware of this and adjust when [this bug is fixed](https://github.com/microsoft/microsoft-ui-xaml/issues/2504).

Basically it is not possible to do it until WinUI3 goes GA. There are still ongoing discussions in WinUI community about basically everything. Since there are a lot of stuff that needs to be implemented, team is changing/adjusting the previous code according to the new APIs.

**Comment by dahovey** [comment]
Any updates on supporting WinUI 3? Is this up for grabs? Is someone already tackling it?

**Comment by dahovey** [comment]
@bkaankose Looks like since your last comment, Windows App SDK (or Project Reunion) for 0.8.5 is GA per [here](https://docs.microsoft.com/en-us/windows/apps/windows-app-sdk/stable-channel). 

I believe I addressed your concerns in PR #4316
